### PR TITLE
fix(swift3): multi-level inheritance support

### PR DESCRIPTION
**Pull request by gierschv**
### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

Supports multi-level inheritance to avoid having the variables from the parents of level > 2

This commit also fixes the comparison of the variables when skipping the variables in the children classes by only comparing their names (in some cases some parent variables were present in children classes)

### CODE REVIEW

**Comment by gierschv** [comment]
cc @wing328 @jaz-ah

**Comment by jaz-ah** [comment]
+1 @wing328

**Comment by wing328** [comment]
@gierschv thanks for the PR. Do you mind sharing the spec so that I can more easily reproduce the issue?

I'm think we may need similar fix in other language generators.

@jaz-ah thanks for reviewing the change.

**Comment by gierschv** [comment]
@wing328 Sure, here is a short example.

3 different views of an object "User":
- `User`: Minimal profile (we can return as a nested object for example)
- `UserPublicProfile`: Full public profile (inherited from `User`)
- `UserDetails`: Private profile with settings & preferences (inherited from `UserPublicProfile`)

```yaml
swagger: "2.0"

info:
  title: Multi-level Inheritance
  version: "v1.0"

schemes: ['http']
consumes: ['application/json']
produces: ['application/json']

paths:
  /me:
    get:
      summary: User account
      responses:
        200:
          description: The user settings
          schema:
            $ref: '#/definitions/UserDetails'

definitions:
  User:
    type: object
    discriminator: userType
    properties:
      id:
        type: string
      username:
        type: string

  UserPublicProfile:
    allOf:
      - $ref: '#/definitions/User'
      - type: object
        properties:
          organization:
            type: string
          htmlUrl:
            type: string

  UserDetails:
    allOf:
      - $ref: '#/definitions/UserPublicProfile'
      - type: object
        properties:
          email:
            type: string
            format: email
```

Without this PR, the model `UserDetails` generated will have the properties of `User`: `id` and `username`, which will not compile since those are already defined in the parent model `User`.

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Swift3Codegen.java [hunk 1]
```diff
@@ -582,7 +590,7 @@ private static CodegenModel reconcileProperties(CodegenModel codegenModel, Codeg
           Iterator<CodegenProperty> iterator = codegenProperties.iterator();
           while (iterator.hasNext()) {
               CodegenProperty codegenProperty = iterator.next();
-              if (codegenProperty.equals(parentModelCodegenProperty)) {
+              if (codegenProperty.baseName == parentModelCodegenProperty.baseName) {
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/Swift3Codegen.java]: So moving forward, the `baseName` of the property becomes the key of the property, which is reasonable as I can't think of a language which allows both parent and child property/member having the same name.

**Pull request merged**